src/DonationForms/AsyncData/Actions/GetAsyncFormDataForListView.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Give\DonationForms\AsyncData\Actions;
+
+use Give\DonationForms\AsyncData\AdminFormListView\AdminFormListViewOptions;
+use Give\DonationForms\AsyncData\AsyncDataHelpers;
+use Give\DonationForms\AsyncData\FormGrid\FormGridViewOptions;
+
+/**
+ * @unreleased
+ */
+class GetAsyncFormDataForListView
+{
+    /**
+     * @unreleased
+     */
+    public function __invoke()
+    {
+        $options = give_clean($_GET);
+
+        if ( ! isset($options['formId'])) {
+            wp_send_json_error(['errorMsg' => __('Missing Form ID.', 'give')]);
+        }
+
+        if ( ! isset($options['nonce']) || ! check_ajax_referer('GiveDonationFormsAsyncDataAjaxNonce', 'nonce')) {
+            wp_send_json_error([
+                'errorMsg' => __('The current user does not have permission to execute this operation.', 'give'),
+            ]);
+        }
+
+        $formId = $options['formId'];
```

Comment:
Should make sure this is an int

```suggestion
        $formId = absint($options['formId']);
```

Reply:
Resolved in ea887b14a050fc16f2c4856cf7eebe8c49d022d7